widget/embedded/src/pages/SelectSwapItemsPage.tsx

```diff
@@ -143,24 +110,19 @@ export function SelectSwapItemsPage(props: PropTypes) {
       />
       <Divider size={16} />
       <TokenList
-        list={filterTokens(tokenList, searchedFor)}
-        selectedBlockchain={selectedBlockchain}
+        list={tokensList}
+        selectedBlockchain={selectedBlockchainName}
         searchedFor={searchedFor}
         onChange={(token) => {
-          const blockchain = blockchains.find(
+          updateToken(token);
+
+          const tokenBlockchain = blockchains.find(
             (chain) => token.blockchain === chain.name
           );
-          if (type === 'from') {
-            setFromToken(token);
-            if (!!blockchain) {
-              setFromBlockchain(blockchain, true);
-            }
-          } else {
-            setToToken(token);
-            if (!!blockchain) {
-              setToBlockchain(blockchain, true);
-            }
+          if (tokenBlockchain) {
+            updateBlockchain(tokenBlockchain);
           }
+
           navigateBackFrom(navigationRoutes.fromSwap);
```

Comment:
@Ikari-Shinji-re I'm not sure it's correct or not, but I thought it should be something like:
``
type === "source" ? navigationRoutes.fromSwap : navigationRoutes.toSwap
``